src/options.c

```diff
@@ -321,46 +321,92 @@ static struct opt_entry opt_list[] = {
 
 #if HAVE_LIBGMP || HAVE_INT128 || HAVE___INT128 || HAVE___INT128_T
 #define PRINCE_USAGE \
-"--prince[=FILE]            PRINCE mode, read words from FILE\n"
+"--prince[=FILE]            PRINCE mode, read words from FILE\n" \
+"--prince-loopback[=FILE]   fetch words from a .pot file\n" \
+"--prince-elem-cnt-min=N    minimum number of elements per chain (1)\n" \
+"--prince-elem-cnt-max=[N]  maximum number of elements per chain (negative N is\n" \
+"                           relative to word length) (8)\n" \
+"--prince-skip=N            initial skip\n" \
+"--prince-limit=N           limit number of candidates generated\n" \
+"--prince-wl-dist-len       calculate length distribution from wordlist\n" \
+"--prince-wl-max=N          load only N words from input wordlist\n" \
+"--prince-case-permute      permute case of first letter\n" \
+"--prince-mmap              memory-map infile (not available with case permute)\n" \
+"--prince-keyspace          just show total keyspace that would be produced\n" \
+"                           (disregarding skip and limit)\n"
 #else
 #define PRINCE_USAGE ""
 #endif
 
-#define JOHN_USAGE	  \
+#ifdef HAVE_FUZZ
+#define FUZZ_USAGE \
+"--fuzz[=DICTFILE]         fuzz formats' prepare(), valid() and split()\n" \
+"--fuzz-dump[=FROM,TO]     dump the fuzzed hashes between FROM and TO to file pwfile.format\n"
+#else
+#define FUZZ_USAGE ""
+#endif
+
+#define JOHN_BANNER	  \
 "John the Ripper " JTR_GIT_VERSION _MP_VERSION DEBUG_STRING ASAN_STRING UBSAN_STRING " [" JOHN_BLD "]\n" \
-"Copyright (c) 1996-2019 by " JOHN_COPYRIGHT "\n" \
+"Copyright (c) 1996-2020 by " JOHN_COPYRIGHT "\n" \
 "Homepage: https://www.openwall.com/john/\n" \
 "\n" \
-"Usage: %s [OPTIONS] [PASSWORD-FILES]\n" \
+"Usage: %s [OPTIONS] [PASSWORD-FILES]\n"                                                                 \
+
```

Comment:
This macro now ends with `\` continuation followed by a blank, which again is fine but a potential foot-gun. While fixing that, you could have the last string in JOHN_BANNER end with double `\n\n` instead of repeatedly following `opt_banner();` with manually printing the second LF.

Reply:
I made the suggested changes directly in Github GUI. If bots buys them, I'll merge now.

BTW I also inlined a single-line-print function only used once, and made another one static. I think that was all.